### ISSUE TITLE
Question: Does Floor offer any form of encryption?

### ISSUE THREAD

**Issue by brettus78** [vitusortner/floor#349]
Hi,

Looking at Floor and it seems like the go-to package, but I really want something that will __encrypt__ the data. 
Does Floor offer this or is it on any road map?

Cheers,
BJ

**Comment by mqus** [comment]
Hey brettus78!
I like your idea. We don't offer it and have not planned anything like it (yet?). 

The general question here is: What is your usecase?
More specific: In what way do you want floor to encrypt *the data*? Do you want to encrypt the stored sqlite file when no one is accessing it (that would be a feature sqflite[1] would have to provide) or do you want to encrypt single fields(That would be your responsibility as a user of floor, because the way things have to be encrypted is somewhat application and context-specific).

Other than that: what kinds of data would you like to encrypt(Passwords, health data, banking information)? Should they only be encrypted on the (flash) storage or also in the memory while the app is running?

I also have to mention that androids and iOS increasingly ~user-hostile~ secure APIs do prevent most accesses to the stored db from third-parties anyway, so regular attackers won't be able to access them (afaik! I don't have a phone with an up2date version so I'm not 100% sure on that)

Last but not least: I'm not a security expert *at all*. I know the usual common advices but that's it. I'm not confident at all to implement something for encryption and be able to call it safe afterwards. If we want to add crypto then my expectations as a contributor are that it has to be reasonably vetted by experts before we can release it.

TLDR: We don't have it, tell us more, I don't want to implement it. :laughing: 

[1][sqflite](https://pub.dev/packages/sqflite) is the library we rely on for homogenous sqlite API on different platforms.

**Comment by brettus78** [comment]
When you put it that way - I have to completely agree. This should really be handled by the code and/or OS.
